core/bug_api.php

```diff
@@ -1545,6 +1550,93 @@ function bug_get_newest_bugnote_timestamp( $p_bug_id ) {
 }
 
 /**
+ * For a list of bug ids, return array of bugnote stats:
+ * the timestamp for the most recent time at which a bugnote
+ * associated with the bug was modified and the total bugnote
+ * count in one db query.
+ * @param array $p_bugs_id Array of Integer representing bug identifiers.
+ * @param integer|null $p_user_id User for checking access levels. null defaults to current user
+ * @return array objects consisting of bugnote stats
+ * @access public
+ * @uses database_api.php
+ */
+function bug_get_bugnote_stats_array( array $p_bugs_id, $p_user_id = null ) {
+	$t_id_array = array();
+	foreach( $p_bugs_id as $t_id ) {
+		$t_id_array[$t_id] = (int)$t_id;
+	}
+	if( empty( $t_id_array ) ) {
+		return array();
+	}
+
+	if ( null === $p_user_id ) {
+		$t_user_id = auth_get_current_user_id();
+	}
+	else {
+		$t_user_id = $p_user_id;
+	}
+
+	db_param_push();
+	$t_params = array();
+	$t_dbparam = array();
+	foreach( $t_id_array as $t_id ) {
+		$t_dbparam[] = db_param();
+		$t_params[] = $t_id;
+	}
+	$t_where_string = ' WHERE n.bug_id in (' . implode( ', ', $t_id_array ) . ')';
+	$t_query = 'SELECT n.id, n.bug_id, n.reporter_id, n.view_state, n.last_modified, n.date_submitted, b.project_id'
+		. ' FROM {bugnote} n JOIN {bug} b ON (n.bug_id = b.id)'
+		. ' WHERE n.bug_id IN (' . implode( ', ', $t_dbparam ) . ')'
+		. ' ORDER BY b.project_id, n.bug_id, n.last_modified';
+	# perform query
+	$t_result = db_query( $t_query, $t_params );
+	$t_query_row = db_fetch_array( $t_result );
+	$t_counter = 0;
+	$t_stats = array();
+	# We need to check for each bugnote if it has permissions to view in respective project.
+	# bugnotes are grouped by project_id and bug_id to save calls to config_get
+	if( $t_query_row ) {
+		do {
```

Comment:
Not sure why you use `if( do { ... } )` here ? 
It seems simpler to do `while( $t_query_row = db_fetch_array( $t_result ) ) { ... }` instead